spec/PeriodicBackgroundSync-index.bs

```diff
@@ -92,3 +94,95 @@ Fetch requests within the <code>periodicsync</code> event while [=in the backgro
 
 ## History Leaking ## {#history-leaking}
 Fetch requests within the <code>periodicsync</code> event while [=in the background=] may reveal something about the client's navigation history to middleboxes on networks different from the one used to create the [=periodicsync registration=]. For instance, the client might visit site https://example.com, which registers a <code>periodicsync</code> event, but based on the implementation, might not fire until after the user has navigated away from the page and changed networks. Middleboxes on the new network may see the fetch requests that the <code>periodicsync</code> event makes. The fetch requests are HTTPS so the request contents will not be leaked but the domain may be (via DNS lookups and IP address of the request). To prevent this leakage of browsing history, the user agent MAY choose to only fire <code>periodicsync</code> events on the network the [=periodicsync registration=] was made on, with the understanding that it will reduce usability by not allowing synchronization opportunistically.
+
+Algorithms {#algorithms}
+=========================
+## <dfn>Calculate a time to fire</dfn> ## {#caculate-time-to-fire}
+This section describes how a user agent can calculate the [=time to fire=] for a [=periodicsync registration=], <var>registration</var>. The time interval between periodicSync events for |registration| MUST be greater than or equal to the [=minInterval=] value associated with |registration|.
+
+The user agent MAY include factors such as user engagement with the origin to decide this time interval, allowing origins with high user engagement to update their web apps more often. In addition, the user agent SHOULD have per-origin and cross-origin caps on this interval, as recommended in [[#privacy]] and Resource Usage sections.
```

Comment:
Should you clarify that the UA may allow aorigins to update more often, as long as it is capped by minInterval?

Reply:
I've already said that minInterval should be a cap on line 101. The suggestion on this line is a recommendation only, so I think it's clear that a user agent can ignore it. Would you disagree?

Reply:
Your call, although one is a recommendation, they feel conflicting to me

Reply:
There's only one requirement here, described just one line above this one, on line 109, which, paraphrased, says that the timeInterval chosen by the user agent must be greater than or equal to minInterval. This gives some recommendations (using SHOULD and MAY) on how to decide this timeInterval. I don't understand how they're contradictory. Can you explain?

Reply:
There's only one requirement here, described just one line above this one, on line 109, which, paraphrased, says that the timeInterval chosen by the user agent must be greater than or equal to minInterval. This gives some recommendations (using SHOULD and MAY) on how to decide this timeInterval. I don't understand how they're contradictory. Can you explain?

Reply:
As I said it's up to you. I find it strange that the spec says you MUST not exceed a limit, then says you MAY exceed this limit under certain assumptions.